keps/sig-network/1453-ingress-api/README.md

```diff
@@ -0,0 +1,1388 @@
+  - N/A
+
+* **Are there any missing metrics that would be useful to have to improve
+  observability if this feature?**
+  Describe the metrics themselves and the reason they weren't added (e.g. cost,
+  implementation difficulties, etc.).
+  - N/A
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+  Think about both cluster-level services (e.g. metrics-server) as well
+  as node-level agents (e.g. specific version of CRI). Focus on external or
+  optional services that are needed. For example, if this feature depends on
+  a cloud provider API, or upon an external software-defined storage or network
+  control plane.
+
+  For each of the fill in the following, thinking both about running user workloads
+  and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - CloudProvider / Ingress Controller
+    - Usage description:
+      - In order for newer features to be utilized like PathType and IngressClass, the 
+      Ingress controller will need to be updated to support these fields and respond 
+      accordingly.
+
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirms the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+  - N/A
+
+* **Will enabling / using this feature result in introducing new API types?**
+  - Yes - this will introduce new types to the Ingress/v1beta1 and Ingress/v1 APIs
+    - Ingress.Spec.Backend.Resource 
+    - Ingress.Spec.IngressClass    
+    - Ingress.Spec.Rules.HTTP.PathType
+  - The supported number of Ingresses per cluster and per namespace will highly depend on the controller implementation 
+    (and on cloud-provider scalability/performance characteristic if applicable). Given this KEP is about the API 
+    itself (and not its implementation), we can't provide any uniformly supportable number here.
+
+* **Will enabling / using this feature result in any new calls to cloud
+  provider?**
+  - Cloud providers will need to update their specific Ingress controller if 
+    they provide one as part of their offering.
+  - New calls will depend on the specific implementation, which is out of scope from this KEP.
+
+* **Will enabling / using this feature result in increasing size or count
+  of the existing API objects?**
+  - There will be new types under networking.k8s.io for Ingress/v1 and Ingress/v1beta1
+      - Ingress.Spec.Backend.Resource 
+      - Ingress.Spec.IngressClass    
+      - Ingress.Spec.Rules.HTTP.PathType
+  - TODO: Mention something about this being affected by the implementation controller and size of cluster
```

Comment:
Actually - I think it won't really depend on controller and cluster size.

The increase in object size will be based on how those new fields will be set (and what features the particular controller implementation supports?).

That said - I wouldn't really bother about it. Ingress is not the object that is being changed super frequently and people don't really have thousands of those. So maybe saying the object size will not increase more than by few hundreds bytes might actually be enough. (Looking into types i'm assuming it's true)